### ISSUE TITLE
Can't cancel registration when finaid approved

### ISSUE THREAD

**Issue by willgearty** [learning-unlimited/ESP-Website#2464]
If a student has already been approved for financial aid, they can not cancel their registration.
It results in an error page that says the following:

> You have already paid for this program! Please contact us directly (using the contact information in the footer of this page) to cancel your registration and to request a refund.

**Comment by bhorowitz** [comment]
What behavior do you want for this? If they cancel their registration should it also cancel the financial aid request?

**Comment by willgearty** [comment]
Well, the default behavior for the cancel registration button for us is to just unenroll the student from all of their classes. So preferably that? It doesn't need to take any action wrt the finaid request.

**Comment by yashikan-dev** [comment]
Hey @willgearty , I would like to work on this issue. Can you please assign it to me?

**Comment by willgearty** [comment]
@yashikan-dev Thanks for your interest in contributing to our project! I've assigned you to this issue. Please make sure to check out our [instructions for getting started](https://github.com/learning-unlimited/ESP-Website/wiki#getting-started) and follow our [contribution rules](https://github.com/learning-unlimited/ESP-Website/wiki#rules). Let me know if you have any questions.

**Comment by yashikan-dev** [comment]
Just as a suggestion, instead of an error, we could implement a branching workflow. If a student who has paid cash clicks 'Cancel,' the system could just show a confirmation page explaining that their class spots will be released immediately. Maybe have a 'Refund Pending' status or notify an admin.
I'd be happy to work on a separate PR for this enhancement once this fix is merged if the maintainers think it's a good direction for the project.

**Comment by willgearty** [comment]
I don't think there's ever a situation where a student has paid in cash _in advance_, only in-person at the actual event.

**Comment by yashikan-dev** [comment]
Okay, thanks for that context! 
In that case, I'll keep the current logic focused on the financial aid fix.
